### PR TITLE
MAINT: regen FUNCTION_LISTING from vega source

### LINKED ISSUE #1411: Documentation bug: expr.format's second, not first argument, is the format specifier
https://github.com/altair-viz/altair/blob/7da67b09383f3c79a8a57fa1dfd428270cc25b31/altair/expr/funcs.py#L63

This matches the documetation but does not work, runs quite a while, and provides loads of NaN

```python
).transform_calculate(
    notweets_humans='format(".2s", datum.notweets)'
)
```

This works fine

```python
).transform_calculate(
    notweets_humans='format(datum.notweets, ".2s")'
)
```

I assume that "first" is not zero-indexed in the documentation, with a "zeroth" argument preceding it ;)

PS. Thanks for bringing _grammar of graphics_ to Python.

### PR DESCRIPTION
**Pull request by genneth**
Semi-manually regenerated from https://github.com/vega/vega/blob/master/docs/docs/expressions.md

Fixes #1411 

Thoughts: 
* No idea how to exhaustively test this
* Don't know if there's a less manual way of doing this --- I literally did some regex and hand-editing to generate the new strings

### CODE REVIEW

**Comment by genneth** [comment]
Hooo boy does python2 and unicode not mix...

File: altair/expr/funcs.py [hunk 1]
```diff
@@ -2,73 +2,139 @@
...
+    "slice": r"Returns a section of _array_ between the _start_ and _end_ indices. If the _end_ argument is negative, it is treated as an offset from the end of the array (_length(array) + end_).",
+    "span": r"Returns the span of _array_: the difference between the last and first elements, or _array[array.length-1] - array[0]_. Or if input is a string: a section of _string_ between the _start_ and _end_ indices. If the _end_ argument is negative, it is treated as an offset from the end of the string (_length(string) + end_)..",
+    # "indexof": r"Returns the first index of _substring_ in the input _string_.",
+    # "lastindexof": r"Returns the last index of _substring_ in the input _string_.",
+    # "length": r"Returns the length of the input _string_.",
```
[the remark this answers is not in the document]
Comment (genneth) [reply altair/expr/funcs.py]: :+1:

**Comment by genneth** [comment]
> Looks good. For testing, I think just double-checking that the update is a strict superset of the previous set of function names would be enough.

And indeed some had been removed. I guess the underlying vega functions were removed? I've put back the ones I took out, but I guess this means that there is the ability to write things which will not execute, depending on the version of vega being use.

**Comment by jakevdp** [comment]
Ah, I didn't realize there had been some removed from the grammar. We may need to split the expr stuff into different versions, so we can keep supporting ``altair.vegalite.v2`` as well as ``altair.vegalite.v3``.

**Comment by genneth** [comment]
> Ah, I didn't realize there had been some removed from the grammar. We may need to split the expr stuff into different versions, so we can keep supporting `altair.vegalite.v2` as well as `altair.vegalite.v3`.

I'll see if I can manage that. Was hoping to quickly tick a box for V3 release, but seem to have unearthed a slightly larger issue!

**Comment by genneth** [comment]
Having stared at the vega source for expressions and functions (and selections?), it's actually not clear that they have ever removed anything. I guess the docs might be incomplete as a source of available expr/functions? That being said, it probably doesn't remove the need for v2/v3 split since v3 will have various expressions which v2 cannot parse?

**Comment by jakevdp** [comment]
> That being said, it probably doesn't remove the need for v2/v3 split since v3 will have various expressions which v2 cannot parse?

Yes, I think that's correct.

**Comment by jakevdp** [comment]
I'm going to go ahead and merge this - thanks!

**Pull request merged**